common/lib/types/message.js

```diff
@@ -137,21 +137,35 @@ var Message = (function() {
 
 	Message.serialize = Utils.encodeBody;
 
-	Message.decode = function(message, options) {
+	Message.decode = function(message, context) {
+		/* The second argument could be either EncodingDecodingContext that contains ChannelOptions or ChannelOptions */
+		if(!context.channelOptions) {
+			var channelOptions = context;
+			context = {
```

Comment:
I don't get why it's valid to create a new context within this function and only ever store it as a local variable, given you set `baseEncodedPreviousPayload` on it just before the function returns, meaning that that value gets immediately discarded. Or am I missing something?

Reply:
Perhaps something using overloads might be neater but I'm not precious about it. At a minimum, however, there could be some commentary explaining this as it's not obvious for maintainers in the future.